File: backend/src/main/java/com/canada/aws/dto/ProductDto.java
package com.canada.aws.dto;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.web.multipart.MultipartFile;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ProductDto {
    private String name;

    private String fullDescription;

    private Date createdTime;

    private Integer quantity;

    private float price;

    private float discountPrice;

    private Date discountStart;

    private Date discountEnd;

    private MultipartFile mainImage;

    private MultipartFile[] extraImages;

    private Integer category_id;

    private Integer brand_id;

}

File: frontend/src/app/model/ProductDto.ts
import { Category } from './Category';
import { Brand } from './Brand';

export class ProductDto {
  name: string;

  fullDescription: string;

  createdTime: Date;

  quantity: number;

  price: number;

  discountPrice: number;

  discountStart: Date;

  discountEnd: Date;

  mainImage: File;

  extraImages: File[];

  category_id: number;

  brand_id: number;

  constructor(){
    this.name = '';
    this.fullDescription = '';
    this.createdTime = new Date();
    this.quantity = null;
    this.price = null;
    this.discountPrice = null;
    this.discountStart = null;
    this.discountEnd = null;
    this.mainImage = null;
    this.extraImages = new Array(8);
    this.category_id = null;
    this.brand_id = null;
  }
}
